search/includes/classes/class-health.php

```diff
@@ -284,6 +297,16 @@ public static function validate_index_posts_count( $options = array() ) {
 
 		$health = new self( $search );
 
+		// Check protected_content feature is enabled before the loop starts
+		$protected_content         = \ElasticPress\Features::factory()->get_registered_feature( 'protected_content' );
+		$protected_content_enabled = $protected_content ? $protected_content->is_active() : false;
+
+		// Add filter to modify query arguments to include password-protected posts if protected_content is not enabled
+		if ( ! $protected_content_enabled ) {
+			add_filter( 'ep_index_posts_args', [ $health, 'query_password_protected_posts' ] );
```

Comment:
When the ProtectedContent feature isn't enabled, we are going to hook the `ep_index_posts_args` filter to call the `query_password_protected_posts` .
It will be fired in the [ES Indexable Post query_db](https://github.com/Automattic/ElasticPress/blob/afd996b92cdf1541c1504af04ef570ed535697ff/includes/classes/Indexable/Post/Post.php#L99)